libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteProgressChangeListener.kt

```diff
@@ -66,14 +68,23 @@ internal class MapRouteProgressChangeListener(
                 job = ThreadController.getMainScopeAndRootJob().scope.launch {
                     val totalDist =
                         (routeProgress.distanceRemaining() + routeProgress.distanceTraveled())
-                    val dist = routeProgress.distanceTraveled() / totalDist
-                    if (dist > 0) {
-                        val deferredExpression = async(Dispatchers.Default) {
-                            routeLine.getExpressionAtOffset(dist)
+                    val percentDistanceTraveled = routeProgress.distanceTraveled() / totalDist
+                    if (percentDistanceTraveled > 0) {
+                        ValueAnimator.ofFloat(lastDistanceValue, percentDistanceTraveled).apply {
+                            duration = ROUTE_LINE_VANISH_ANIMATION_DURATION
+                            interpolator = LinearInterpolator()
+                            startDelay = ROUTE_LINE_VANISH_ANIMATION_DELAY
+                            addUpdateListener {
+                                val animationDistanceValue = it.animatedValue as Float
+                                if (animationDistanceValue > MINIMUM_ROUTE_LINE_OFFSET) {
+                                    val expression = routeLine.getExpressionAtOffset(animationDistanceValue)
+                                    routeLine.hideShieldLineAtOffset(animationDistanceValue)
+                                    routeLine.decorateRouteLine(expression)
+                                }
+                            }
+                            start()
```

Comment:
Does the `ValueAnimator`'s `end()` method need to be called anywhere? Tied to the destruction of the `MapRouteLine`, map, etc.?

Reply:
I don't think so since only lives for a second or so and isn't scoped to the class. I haven't used ValueAnimator a lot but I didn't see end() being called in the Android documentation.

Reply:
👍 Was on my mind after all of [my pulsing animator work](https://github.com/mapbox/mapbox-gl-native-android/pull/172) 😄

Reply:
I'm late to the party, but this is a good lead @langsmith - https://github.com/mapbox/mapbox-navigation-android/issues/3030.